importers/yelp.ca/generate_csv.py

```diff
@@ -0,0 +1,127 @@
+        review = root.cssselect('.review-content > p')
+        number_of_reviews = Counter(review) # retreives list with elements
+        misc_info = None
+        wheelchair = None
+        accessible = None
+        for i in range(len(number_of_reviews)):
+            reviews = review[i].xpath('string()')
+            # loops through all reviews
+            wheelchair = re.findall(r"([^.]*?wheelchair[^.]*\.)",reviews)
+            # searches for keyword in reviews and returns list containing sentences containing keyword
+            accessible = re.findall(r"([^.]*?accessible[^.]*\.)",reviews)
+
+        business_info = get_text_from_css(root,'div.ywidget > ul')
+        business_info = business_info.replace(" ","").split('\n')
+        filtered = []
+        for i in business_info:
+            if i != "":
+                filtered.append(i)
+        dict_business_info = dict(itertools.izip_longest(*[iter(filtered)] * 2, fillvalue = "" ))
+        # looking for comments with information regarding accessibility incase wheelchair accessible is empty or absent        if dict_business_info['WheelchairAccessible'] == "Yes":
+        if 'WheelchairAccessible' in filtered:
+            wheelchair_accessible = dict_business_info['WheelchairAccessible']
+        elif wheelchair:
+            misc_info = wheelchair[0]
+        elif accessible:
+            misc_info = accessible[0]
+        else:
+            wheelchair_accessible = "N/A"
+            misc_info = "Call restaurant/store to verify accessibility."
+        if "Wi-Fi" in filtered:
+            wifi = dict_business_info['Wi-Fi']
+        else:
+            wifi = "N/A"
+        if "Parking" in filtered:
+            parking = dict_business_info['Parking']
+        else:
+            parking = "N/A"
+        return {
+            'name': biz_name,
+            'address': address,
+            'longitude': longitude,
+            'latitude': latitude,
+            'wheelchair accessible': wheelchair_accessible,
+            'wifi': wifi,
+            'phone number': phone_number,
+            'parking': parking,
+            'map': yelp_base_url + location_url,
+            'Misc Info': misc_info
+            }
+
+
+def get_all_downloaded_locations():
+    """
+    loops through directory and returns locations list with filenames
+    that begin with raw_html
+    """
+    locations = []
+    data_dir = 'data/'
+    for (dirname, dirs, files) in os.walk('.'):
+        for filename in files:
+            if filename.startswith('raw_html'):
+                filename = data_dir + filename
+                location_details = get_location_info(filename)
+                locations.append(location_details)
+    return locations
+
+
+def generate_csv():
+    print "Gathering info.."
+    locations = get_all_downloaded_locations()
+    keys = ['name','address','latitude','longitude','wheelchair accessible','wifi','phone number','parking', 'map', 'Misc Info']
+    for i in range(len(locations)):
```

Comment:
You should be making 1 CSV file that has a separate line/row for each location.
Instead, you're making a different CSV file for each location.

You should swap the "with" statement around with this for loop and just make one CSV file.  It could be named something like locations.csv.